superset-frontend/src/views/CRUD/data/database/DatabaseModal/index.tsx

```diff
@@ -566,6 +602,93 @@ const DatabaseModal: FunctionComponent<DatabaseModalProps> = ({
 
   const [useSSHTunneling, setUseSSHTunneling] = useState<boolean>(false);
 
+  const createPostProcessingCallbacks = useRef<any>({});
+  const updatePostProcessingCallbacks = useRef<any>({});
```

Comment:
If I understand this correctly, it looks like the child is sending a function from a lower level component up to the a higher component by attaching it to a ref, is that right? Would it be easier to import the extension into this parent index component, and then you can call the create and update callbacks later in the parent and then pass down just the visual components to the ExtraOptions? That way you don't have to pass the callbacks up to the parent. I'm not sure if that's necessarily the solution, but I know that these extension points are complicated, so trying to help suggest a simpler solution if there is one.